### PR TITLE
RDK-588: GC/bmalloc finetuning on Broadcom platforms

### PR DESCRIPTION
**Pull request by tomasz-karczewski-red**
Several changes making it possible to better control GC, especially on Broadcom platforms:

- some IOS branches related to GC applied also for BROADCOM cases
- MemoryPressureHandler threshold made configurable with envs: WPE_MEMORY_PRESSURE_HANDLER_CONSERVATIVE_THRESHOLD_FRACTION WPE_MEMORY_PRESSURE_HANDLER_STRICT_THRESHOLD_FRACTION
- bmalloc memoryPressureThreshold made configurable with WPE_BMALLOC_MEMORY_PRESSURE_THRESHOLD env
- bmalloc scavengerBytesPerMemoryPressureCheck configurable with WPE_BMALLOC_SCAVENGER_BYTES_PER_MEMORY_PRESSURE_CHECK env
- garbageCollectJavaScriptObjects: also use shrinkFootprintWhenIdle, to improve memory cleanup
- bmalloc should take WPE_RAM_SIZE env into account when determining total available memory

### CODE REVIEW

File: Source/WTF/wtf/MemoryPressureHandler.cpp [hunk 1]
```diff
@@ -484,8 +493,8 @@ void MemoryPressureHandler::setDispatchQueue(OSObjectPtr<dispatch_queue_t>&& que
 MemoryPressureHandler::Configuration::Configuration()
     : baseThreshold(std::min(3 * GB, ramSize()))
     , baseThresholdVideo(1 * GB)
-    , conservativeThresholdFraction(s_conservativeThresholdFraction)
-    , strictThresholdFraction(s_strictThresholdFraction)
+    , conservativeThresholdFraction(from_env_or_default("WPE_MEMORY_PRESSURE_HANDLER_CONSERVATIVE_THRESHOLD_FRACTION",s_conservativeThresholdFraction))
+    , strictThresholdFraction(from_env_or_default("WPE_MEMORY_PRESSURE_HANDLER_STRICT_THRESHOLD_FRACTION", s_strictThresholdFraction))
     , killThresholdFraction(s_killThresholdFraction)
```
Comment (magomez) [inline Source/WTF/wtf/MemoryPressureHandler.cpp:498]: You can set the conservative and strict fractions through the MemoryPressureHandlerConfiguration API. These new env vars are not needed, please use the API

Comment (tomasz-karczewski-red) [reply Source/WTF/wtf/MemoryPressureHandler.cpp:498]: sure - will remove this part

**Review by magomez — CHANGES_REQUESTED** [review]
Please, split this change into several commits. This is touching many things in many places that should be independent commits.
Also, as I commented before, there's the MemoryPressureHandlerConfiguration API to set the config from the app. Please use it and don't change the default values or add new env vars to overwrite them.

File: Source/JavaScriptCore/heap/FullGCActivityCallback.cpp [hunk 2]
```diff
@@ -41,7 +41,7 @@ void FullGCActivityCallback::doCollection(VM& vm)
     Heap& heap = vm.heap;
     m_didGCRecently = false;
 
-#if !PLATFORM(IOS_FAMILY) || PLATFORM(MACCATALYST)
+#if (!PLATFORM(IOS_FAMILY) && !PLATFORM(BROADCOM)) || PLATFORM(MACCATALYST)
```
Comment (modeveci) [inline Source/JavaScriptCore/heap/FullGCActivityCallback.cpp:44]: This should be a separate PR and we need to get feedback from other users. And if it is necessary, we need to think use WPE flag instead of a vendor.

File: Source/WTF/wtf/MemoryPressureHandler.cpp (hunk 1 above)
Comment (modeveci) [reply Source/WTF/wtf/MemoryPressureHandler.cpp:498]: As stated by @magomez , you can make changes around these settings from rdkservices:
https://github.com/rdkcentral/rdkservices/blob/sprint/24Q4/WebKitBrowser/WebKitImplementation.cpp#L500

**Comment by modeveci** [comment]
I agree with @magomez , granularity of commits should be limited to the single fix(feature). And it requires other Broadcom users' reviews. Preferable we would like to have change based on WPE flag rather than Broadcom flag

**Comment by tomasz-karczewski-red** [comment]
thanks for the review @magomez @modeveci - I will submit this into smaller commits and remove the WPE_MEMORY_PRESSURE_HANDLER_CONSERVATIVE_THRESHOLD_FRACTION part.
